lib/services/upstart.c

```diff
@@ -311,18 +311,28 @@ get_first_instance(const gchar * job, int timeout)
     return instance;
 }
 
+/*!
+ * \internal
+ * \brief Parse result of Upstart status check
+ *
+ * \param[in] name      DBus interface name for property that was checked
+ * \param[in] state     Property value
+ * \param[in] userdata  Status action that check was done for
+ */
 static void
-upstart_job_check(const char *name, const char *state, void *userdata)
+parse_status_result(const char *name, const char *state, void *userdata)
 {
-    svc_action_t * op = userdata;
+    svc_action_t *op = userdata;
 
-    if (state && g_strcmp0(state, "running") == 0) {
+    if (pcmk__str_eq(state, "running", pcmk__str_none)) {
         op->rc = PCMK_OCF_OK;
+        op->status = PCMK_EXEC_DONE;
```

Comment:
could pull that ahead of the if

Reply:
intentional -- a later PR will functionize setting them together, so this will help make that one easier to review